client/state/selectors/is-requesting-jetpack-onboarding-settings.js

```diff
@@ -0,0 +1,28 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { getRequest } from 'state/selectors';
+import { requestJetpackOnboardingSettings } from 'state/jetpack-onboarding/actions';
+
+/**
+ * Returns true if we are currently making a request to fetch the Jetpack settings. False otherwise
+ *
+ * @param  {Object}  state       Global state tree
+ * @param  {Number}  siteId      The ID of the site we're querying
+ * @param  {Object}  query       An optional query to be passed to the JP settings endpoint
+ * @return {Boolean}             Whether Jetpack settings are currently being requested
+ */
+export default function isRequestingJetpackOnboardingSettings( state, siteId, query ) {
```

Comment:
🤔 I wonder how common this is and whether some abstraction would make sense, like `isDataLayerRequestInFlight( request )`

Reply:
Indeed, but that's what the `getRequest()` used in the implementation does, right? Unfortunately, that still reads a bit clunky, since you have to specify the entire action, including args.

I would've been okay with _not_ introducing the `isRequesting...` selector here and just continuing to use `getRequest( state, action ).isLoading`, but the `isRequesting...` thingy will come in really handy when replacing the existing JP settings one (#23393).